pos_sale_order_delivery_carrier/static/src/js/Popups/CreateOrderPopup.esm.js

```diff
@@ -0,0 +1,90 @@
+/** @odoo-module **/
+
+import CreateOrderPopup from "point_of_sale.CreateOrderPopup";
+import Registries from "point_of_sale.Registries";
+import framework from "web.framework";
+
+const {useState} = owl;
+
+const PosDeliveryCarrierCreateOrderPopup = (CreateOrderPopup) =>
+    class extends CreateOrderPopup {
+        // @override
+        setup() {
+            super.setup();
+            this.state = useState({addDeliveryCarrier: false});
+        }
+        get currentOrder() {
+            return this.env.pos.get_order();
+        }
+        onClickToShipping() {
+            this.state.addDeliveryCarrier = !this.state.addDeliveryCarrier;
+        }
+        // @override
+        async _createSaleOrder(order_state) {
```

Comment:
Since `pos_order_to_sale_order` is also a module of this repository, and we have control over it, it'd be better to improve this module in order to make the method extendable without having to do a complete override.